### DISCUSSION TITLE
Performance, Features comparison chart

### DISCUSSION THREAD

**Discussion by mkohlhaas** [dankamongmen/notcurses#2157] [Q&A]
Frigging project!!! Just frigging awesome!!! Thanks a lot for it!!!

- Is there a performance and features comparison table for different terminals available?
- What is your preferred terminal when it comes to performance?

Thanks again! 🙏 😊

**Comment by dankamongmen** [comment]
hey! thanks for your kind words.

i have some old data (late 2019) on my wiki: https://nick-black.com/dankwiki/index.php?title=Notcurses#Terminal_emulators
i have some less old data in appendix B of my book: https://nick-black.com/htp-notcurses.pdf

but i have not run systematic cross-terminal benchmarks recently, no. i ought!

regarding performance, foot, kitty, alacritty, and wezterm are all right up there. contour does some things very quickly, and some things less so. the others mentioned seem pretty fast at everything.

kitty graphics and sixel graphics are very different. kitty applies zlib compression, which introduces some definite latency, but can also cut bandwidth such that it ends up faster than sixel, especially over a network. when we add sideband kitty graphics, that'll cut kitty drastically, but only for local sessions. kitty gives a better picture than sixel, so it's not apples-to-oranges anyway.

alacritty is very speedy, but can't do emoji as far as i know.

i currently use kitty as my primary X.org terminal emulator, and foot as my primary Wayland emulator. when i can build Wezterm on my debian workstation, i throw a good bit of it in, as well.

for any given scenario, xterm tends to not be faster than other terminals i test. sometimes it's up near the rest, sometimes very far back.

mlterm, VTE, and Konsole, are outperformed by the others i've listed, in my testing.

i don't use anything but Linux for my own machines, so i don't have much of a feel for e.g. iTerm2.
